ndk/src/native_window.rs

```diff
@@ -126,4 +126,61 @@ impl NativeWindow {
     pub unsafe fn to_surface(&self, env: *mut JNIEnv) -> jobject {
         ffi::ANativeWindow_toSurface(env, self.ptr().as_ptr())
     }
+
+    /// Lock the window's next drawing surface for writing.
+    ///
+    /// Optionally pass the region you intend to draw into `dirty_bounds`.  When this function
+    /// returns it is updated (commonly enlarged) with the actual area the caller needs to redraw.
+    pub fn lock(
+        &self,
+        dirty_bounds: Option<&mut ffi::ARect>,
+    ) -> Result<NativeWindowBufferLockGuard> {
+        let dirty_bounds = match dirty_bounds {
+            Some(dirty_bounds) => dirty_bounds,
+            None => std::ptr::null_mut(),
+        };
+        let mut out = unsafe { std::mem::zeroed() };
+        let ret = unsafe { ffi::ANativeWindow_lock(self.ptr.as_ptr(), &mut out, dirty_bounds) };
+        status_to_io_result(ret, NativeWindowBufferLockGuard(out, self))
+    }
+}
+
+/// Lock holding the next drawing surface for writing.  It is unlocked and posted on [`drop()`].
+#[derive(Debug)]
+pub struct NativeWindowBufferLockGuard<'a>(ffi::ANativeWindow_Buffer, &'a NativeWindow);
+
+impl<'a> NativeWindowBufferLockGuard<'a> {
+    /// The number of pixels that are shown horizontally.
+    pub fn width(&self) -> i32 {
```

Comment:
seems like width, height and stride could be exposed as `usize` in Rust

Reply:
Indeed, that makes it easier to work with in loops and `ops::Idex`.